### ISSUE TITLE
getInitialProps gets called multiple times for AMP

### ISSUE THREAD

**Issue by getanwar** [zeit/next.js#9111]
# Bug report

## Describe the bug

Codes in `getInitialProps` get called multiple times if the component is AMP. I tried installing a fresh `nextjs` project and faced the same behavior.

## To Reproduce

```js
export const config = { amp: true };

const Home = () => (
   // ...
);

Home.getInitialProps = () => {
    console.log("This is getting called so many times");
    return {
        a: 1 // return a valid object
    };
};

export default Home;

```

Now open `home` in a browser and look at the terminal. I cleared the terminal and reloaded the page again but the result is same. There are multiple console logs (at least 6 for me!).

## Expected behavior

Should get called only once.

## Screenshots

<img width="402" alt="Screen Shot 2019-10-17 at 5 04 36 PM" src="https://user-images.githubusercontent.com/6163268/67016096-4d425e80-f100-11e9-86ab-e97d4142ada0.png">

## System information

- OS: macOS (Mojave)
- Browser: Chrome (latest)
- Version of Next.js: 9.1.1

## Additional context

There is a warning in the terminal but no message is shown!

**Comment by getanwar** [comment]
### Additional information

If it was only occurred in `dev` then I would have guessed HRM is causing it. But it's happening in `build` also. Here is a screenshot:

<img width="315" alt="Screen Shot 2019-10-17 at 5 10 15 PM" src="https://user-images.githubusercontent.com/6163268/67016623-26385c80-f101-11e9-9ca1-4a2dcf1eb101.png">

**Comment by ijjk** [comment]
Hi, I'm not able to reproduce this occurring in a productino build with a minimal project using Next.js v9.1.1. (see below) 

![Kapture 2019-10-17 at 16 51 44](https://user-images.githubusercontent.com/22380829/67050682-af9a5f80-f0fe-11e9-89c3-83b4f3153cce.gif)

It might be called a few of times if you leave your browser open and restart the dev server but that is expected since we re-open the HMR connection and reload the page if you restarted the server.

I'm gonna close this since it doesn't appear to be a bug. If you are able to reproduce this in production please reply with additional info

**Comment by getanwar** [comment]
Seems like you haven't opened the page in a browser. 
Would you please try again by doing so:
`npm start` > clear the terminal > open the page in a browser > get back to the terminal to see the console

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
